### PR TITLE
Fix C99 `I` not working with NumPy 2.0.1

### PR DESCRIPTION
**Pull request by sklam**
Workaround problem with `I`

### CODE REVIEW

**Comment by sklam** [comment]
BFID: `numba_smoketest_cpu_yaml_201`

**Comment by sklam** [comment]
`numba_smoketest_cpu_yaml_201` not finished yet with test jobs but all build jobs have passed

**Comment by tiran** [comment]
According to [The Open Group Base Specifications Issue 8](https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/complex.h.html), you have to check for `_Imaginary_I` first and then fall back to `_Complex_I`, see https://github.com/numpy/numpy/issues/27083#issuecomment-2267389934

File: numba/_helperlib.c [hunk 1]
```diff
@@ -19,7 +19,13 @@
 #else
     #include <stdint.h>
     #define _complex_float_t complex float
-    #define _complex_float_ctor(r, i) (r + I * i)
+    #if defined(_Imaginary_I)
+        #define _complex_float_ctor(r, i) (r + _Imaginary_I * i)
+    #elif defined(_Complex_I)
+        #define _complex_float_ctor(r, i) (r + _Complex_I * i)
+    #else
+        #error "Lack _Imaginery_I and _Complex_I"
```
Comment (gmarkall) [inline numba/_helperlib.c:27]: A small typo that might have a user grepping for something that will never be found:

```suggestion
        #error "Lack _Imaginary_I and _Complex_I"
```

**Review by gmarkall — CHANGES_REQUESTED** [review]
LGTM apart from the typo.

**Review by gmarkall — APPROVED** [review]

**Comment by sklam** [comment]
CI problem fixed

**Pull request merged**